packages/server/src/lib/targeting.ts

```diff
@@ -124,11 +124,15 @@ export const pageContextMatches = (
 ): boolean => {
     const { tagIds, sectionIds, excludedTagIds, excludedSectionIds } = testTargeting;
 
-    return (
-        (tagIds.length === 0 || pageHasATag(tagIds, pageContext.tagIds)) &&
-        (sectionIds.length === 0 || pageHasASection(sectionIds, pageContext.sectionId)) &&
-        (excludedTagIds.length === 0 || !pageHasATag(excludedTagIds, pageContext.tagIds)) &&
-        (excludedSectionIds.length === 0 ||
-            !pageHasASection(excludedSectionIds, pageContext.sectionId))
-    );
+    const inclusionsMatch: boolean =
+        (tagIds.length === 0 && sectionIds.length === 0) || // no targeting
+        pageHasATag(tagIds, pageContext.tagIds) ||
+        pageHasASection(sectionIds, pageContext.sectionId);
+
+    const exclusionsMatch: boolean =
```

Comment:
I like this approach of breaking this into separate steps of matching exclusions and then negating with `!`.